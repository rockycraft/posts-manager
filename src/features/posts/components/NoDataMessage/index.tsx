import { Center, Text } from "@chakra-ui/react";
import { memo } from "react";

function NoDataMessage() {
	return (
		<Center>
			<Text data-testid="no-data-message" textStyle="2xl" fontWeight="bold">
				No hay posts para mostrar
			</Text>
		</Center>
	);
}

export default memo(NoDataMessage);
