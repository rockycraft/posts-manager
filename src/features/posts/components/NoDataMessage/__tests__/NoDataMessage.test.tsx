import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import NoDataMessage from "../index";
import { describe, it, expect, vi } from "vitest";
import { Provider } from "@/components/ui/provider";

Object.defineProperty(window, "matchMedia", {
	writable: true,
	value: vi.fn().mockImplementation((query) => ({
		matches: false,
		media: query,
		onchange: null,
		addListener: vi.fn(), // deprecated
		removeListener: vi.fn(), // deprecated
		addEventListener: vi.fn(),
		removeEventListener: vi.fn(),
		dispatchEvent: vi.fn(),
	})),
});

describe("NoDataMessage component", () => {
	it("should render the no data message", () => {
		render(
			<Provider>
				<NoDataMessage />
			</Provider>,
		);

		expect(screen.getByTestId("no-data-message")).toBeInTheDocument();
	});

	it("should match snapshot", () => {
		const { asFragment } = render(
			<Provider>
				<NoDataMessage />
			</Provider>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
