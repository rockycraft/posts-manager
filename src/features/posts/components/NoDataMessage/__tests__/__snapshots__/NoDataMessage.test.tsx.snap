// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoDataMessage component > should match snapshot 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,i,s,u,m,a,l,h)=&gt;{let d=document.documentElement,w=["light","dark"];function p(n){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&a?m.map(f=&gt;a[f]||f):m;k?(d.classList.remove(...S),d.classList.add(a&&a[n]?a[n]:n)):d.setAttribute(y,n)}),R(n)}function R(n){h&&w.includes(n)&&(d.style.colorScheme=n)}function c(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(u)p(u);else try{let n=localStorage.getItem(i)||s,y=l&&n==="system"?c():n;p(y)}catch(n){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <div
    class="css-d82rsv"
  >
    <p
      class="css-1bkcq3"
      data-testid="no-data-message"
    >
      No hay posts para mostrar
    </p>
  </div>
</DocumentFragment>
`;
